### ISSUE TITLE
Streamline food list interactions

### ISSUE THREAD

**Issue by janibonnevier** [davidhealey/waistline#439]
## Issue

When attempting to select an item in the food list for insertion into the diary, I often miss the checkbox with my finger, which takes me to the food editing view. It is frustrating that the most common operation requires hitting such a small target (it's also on the far side of the screen, for a right-handed person trying to use the app with one hand).

## Proposal

My view is that the more frequent an operation is, the easier it should be to do in the UI. I assume that the food list operations ordered from most frequent to least, are as follows:

1. Select for diary
2. Edit
3. Delete

So I propose that

- Short press anywhere on the food selects it.
- Long press anywhere on the food opens it for editing.
- Deletion is an option in the editing view.

**Comment by janibonnevier** [comment]
> Do you have a particularly wide phone? I use Waistline with just my right hand every day and don't have any issues reaching my thumb across to press the checkbox

I'm not sure. But sizes of phones, hands and fingers vary, as does hand dexterity. My parenthesis about one-handed use was not my main point. I was more concerned about the size of the target, regardless of the number of hands being used.

> It's a definite no to adding the delete option in the edit view, that view is already getting crowded. A swipe action might be a good way but that generally is a two handed action - one hand to hold the phone, one finger on the other hand to swipe.

Fair enough. Could it be OK that it's a two-handed action considering that it presumably is not that common? I worry that it might be difficult to discover.

Other alternatives for consideration:

- An Edit button (e.g. :pencil2:) on each item, similar in size to the current checkbox. Deletion as is today.
  
  This is, in a sense, the current interactions but with Select and Edit swapped according to their priority. Having both a checkbox (for visual indication of selection) and an edit button might make the food list crowded. Could selection be indicated in some other way, like color, or a more compact indicator?
- Long press opens a context menu with options for Edit and Delete.
  
  This of course makes both of them more cumbersome to access, but it's a tradeoff that I would be willing to make for easier selection.

> What about an option to place the checkbox on the right side of the screen?

It could be worth a shot if it's significantly easier to implement.

Related reading: https://en.wikipedia.org/wiki/Fitts's_law

**Comment by davidhealey** [comment]
> I worry that it might be difficult to discover.

I don't think it would be any more difficult than tap and hold. I think swiping is quite a common gesture in apps.







>  Long press opens a context menu with options for Edit and Delete.

This is how it worked in Waistline 2, it wasn't so popular. 

Maybe the best option is the simplest, let's make the checkbox hit area larger (I need to do this for some of the + buttons too). Do you use K9 mail? It has a larger hit area on the left-hand side, I think that might be a good model. My thinking is we'll basically make any taps that are vaguely in the vicinity of the checkbox register as a hit.

**Comment by janibonnevier** [comment]
Sounds good.

(But let me just point out that in K9, selection is not the number 1 most common action like it is in the food list.)

**Comment by h-h-h-h** [comment]
"Edit" should still be easy to reach. I wouldn't like having to go over a context menu for that. What about invisibly splitting the entries in half and doing "Select" on the left and "Edit" on the right. When the user does item tapping for the first time ever, an informational dialog could appear, shortly explaining the behavior and allowing the user to hide the message once and for all.

There also could be an "Edit" icon on the right side, no informational dialog, but still the function-splitting invisible line in the center.
